crates/sui-kvstore/src/bigtable/client.rs

```diff
@@ -0,0 +1,411 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::bigtable::proto::bigtable::v2::bigtable_client::BigtableClient as BigtableInternalClient;
+use crate::bigtable::proto::bigtable::v2::mutate_rows_request::Entry;
+use crate::bigtable::proto::bigtable::v2::mutation::SetCell;
+use crate::bigtable::proto::bigtable::v2::read_rows_response::cell_chunk::RowStatus;
+use crate::bigtable::proto::bigtable::v2::{
+    mutation, MutateRowsRequest, MutateRowsResponse, Mutation, ReadRowsRequest, RowSet,
+};
+use crate::{Checkpoint, KeyValueStoreReader, KeyValueStoreWriter, TransactionData};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use gcp_auth::{Token, TokenProvider};
+use http::{HeaderValue, Request, Response};
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::{Arc, RwLock};
+use std::task::{Context, Poll};
+use std::time::Duration;
+use sui_types::base_types::{SequenceNumber, TransactionDigest};
+use sui_types::full_checkpoint_content::CheckpointData;
+use sui_types::object::Object;
+use sui_types::storage::ObjectKey;
+use tonic::body::BoxBody;
+use tonic::codegen::Service;
+use tonic::transport::{Certificate, Channel, ClientTlsConfig};
+use tonic::Streaming;
+use tracing::error;
+
+const OBJECTS_TABLE: &str = "objects";
+const CHECKPOINTS_TABLE: &str = "checkpoints";
+const TRANSACTIONS_TABLE: &str = "transactions";
+
+const COLUMN_FAMILY_NAME: &str = "sui";
+const DEFAULT_COLUMN_QUALIFIER: &str = "";
+const CHECKPOINT_SUMMARY_COLUMN_QUALIFIER: &str = "s";
+const CHECKPOINT_CONTENTS_COLUMN_QUALIFIER: &str = "c";
+const TRANSACTION_COLUMN_QUALIFIER: &str = "tx";
+const EFFECTS_COLUMN_QUALIFIER: &str = "ef";
+const EVENTS_COLUMN_QUALIFIER: &str = "ev";
+const TIMESTAMP_COLUMN_QUALIFIER: &str = "ts";
+const CHECKPOINT_NUMBER_COLUMN_QUALIFIER: &str = "cn";
+
+type Bytes = Vec<u8>;
```

Comment:
Don't necessarily need to worry about changing this in this PR, but I think it would be a win to possibly change this, and all protobuf bytes fields, to the `bytes::Bytes` type. This can be done with https://docs.rs/prost-build/latest/prost_build/struct.Config.html#method.bytes for the protobuf generation.